server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -112,7 +112,7 @@ public static boolean isFailureStoreFeatureFlagEnabled() {
     private final IndexMode indexMode;
     @Nullable
     private final DataStreamLifecycle lifecycle;
-    private final boolean failureStoreEnabled;
+    private final DataStreamOptions dataStreamOptions;
```

Comment:
I thought it would be easier to refer to an empty constant than always perform the null check before accessing this object. Only on serialisation we use null to avoid serialiazing empty configurations.